pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -281,3 +282,12 @@ var ClusterResource = k8sutil.CustomResource{
 	Kind:       reflect.TypeOf(cephv1.CephCluster{}).Name(),
 	APIVersion: fmt.Sprintf("%s/%s", cephv1.CustomResourceGroup, cephv1.Version),
 }
+
+// RecoverAndLogException handles and logs panics from a controller Reconcile loop.
+func RecoverAndLogException() {
+	if r := recover(); r != nil {
+		logger.Errorf("Panic: %v", r)
+		logger.Errorf("Stack trace:\n%s", string(debug.Stack()))
+		panic(r)
```

Comment:
From [this comment](https://github.com/rook/rook/pull/16150#discussion_r2231472605) let's remove the panic, but we can discuss in huddle if needed
```suggestion
```

Reply:
Seems okay to me. It's a good point that panics are unlikely to resolve on the next reconcile, so this can save some log churn